bundles/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/netutils/WebSocketConnection.java

```diff
@@ -73,25 +87,55 @@ public void start(String ip) {
             return;
         } else if (connectionState == ConnectionState.DISCONNECTING) {
             logger.warn("{} trying to re-connect while still disconnecting", socketName);
+            return;
         }
         try {
+            connectionState = ConnectionState.CONNECTING;
             URI destUri = URI.create("ws://" + ip);
             client.start();
             logger.debug("Trying to connect {} to {}", socketName, destUri);
             client.connect(this, destUri).get();
         } catch (Exception e) {
-            connectionListener.connectionLost("Error while connecting: " + e.getMessage());
+            String reason = "Error while connecting: " + e.getMessage();
+            if (e.getMessage() == null) {
+                logger.warn("{}: {}", socketName, reason, e);
+            } else {
+                logger.warn("{}: {}", socketName, reason);
+            }
+            connectionListener.webSocketConnectionLost(reason);
```

Comment:
Something seems a little fishy here - did you mean `e.getCause()`? If `e.getMessage()` is null, then `reason` will be "Error while connecting: null", is this intended the be logged like this and with full call stack? Just checking. 🙂

Reply:
It's intended for the message to the `connectionListener`. IMO it doesn't hurt when we log the full exception and always using `reason` is a bit clearer than creating different messages for all cases.